pkg/agent/agent.go

```diff
@@ -70,36 +63,41 @@ func (a *Agent) run() error {
 		return err
 	}
 
-	bundle, err := a.loadBundle()
-	if err != nil {
-		return err
-	}
-
-	svid, key, err := a.loadSVID()
+	as, err := a.attest()
 	if err != nil {
 		return err
 	}
 
-	if svid == nil {
-		svid, bundle, err = a.newSVID(key, bundle)
-		if err != nil {
-			return err
-		}
-	}
-
-	err = a.startManager(svid, key, bundle)
+	err = a.startManager(as.SVID, as.Key, as.Bundle)
 	if err != nil {
 		return err
 	}
-
-	a.t.Go(func() error { return a.startEndpoints(bundle) })
+	a.t.Go(func() error { return a.startEndpoints(as.Bundle) })
 	a.t.Go(a.superviseManager)
 
 	<-a.t.Dying()
 	a.shutdown()
 	return nil
 }
 
+func (a *Agent) startPlugins() error {
+	return a.Catalog.Run()
+}
+
+func (a *Agent) attest() (*attestor.AttestationResult, error) {
+	config := attestor.Config{
+		Catalog:       a.Catalog,
+		JoinToken:     a.c.JoinToken,
+		TrustDomain:   a.c.TrustDomain,
+		TrustBundle:   a.c.TrustBundle,
+		DataDir:       a.c.DataDir,
+		Log:           a.c.Log.WithField("subsystem_name", "attestor"),
+		ServerAddress: a.c.ServerAddress,
+	}
+	a.Attestor = attestor.New(&config)
```

Comment:
Do we need to store this on agent struct? Or can we get away with `return attestor.New(&config).Attest()`?